### PR TITLE
Update re2, absl, and googletest

### PR DESCRIPTION
**Pull request by jonmeow**
re2 has migrated from StringPiece to absl::string_view since we last updated. There are also some changes to how string_view is packaged by absl. As I do test work, I'm noticing these gaps and think an update would help.

### CODE REVIEW

**Comment by jonmeow** [comment]
Note the relevant commit here is https://github.com/carbon-language/carbon-lang/pull/3042/commits/373d8f2b6680b6b9037742251f54734f2542ebe5, but if this is approved before #3035 I'll probably rebase. Right now they're stacked because I'm working on changes needing both.

**Review by chandlerc — APPROVED** [review]
LG, leave it to you whether to rebase and merge or wait for the earlier commit and merge.

**Pull request merged**
